benchmark/fri/fri_runner.h

```diff
@@ -0,0 +1,92 @@
+#ifndef BENCHMARK_FRI_FRI_RUNNER_H_
+#define BENCHMARK_FRI_FRI_RUNNER_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+// clang-format off
+#include "benchmark/fri/fri_config.h"
+#include "benchmark/simple_reporter.h"
+// clang-format on
+#include "tachyon/base/time/time.h"
+#include "tachyon/c/math/finite_fields/baby_bear/baby_bear_type_traits.h"
+#include "tachyon/crypto/commitments/fri/two_adic_multiplicative_coset.h"
+#include "tachyon/math/matrix/matrix_types.h"
+
+namespace tachyon::benchmark {
+
+template <typename PCS>
+class FRIRunner {
+ public:
+  using F = typename PCS::F;
+  using Domain = crypto::TwoAdicMultiplicativeCoset<F>;
+
+  typedef tachyon_baby_bear* (*ExternalFn)(const tachyon_baby_bear* data,
+                                           const size_t* degrees,
+                                           size_t num_of_degrees,
+                                           size_t batch_size,
+                                           uint32_t log_blowup,
+                                           uint64_t* duration);
+
+  FRIRunner(SimpleReporter& reporter, const FRIConfig& config, PCS& pcs)
+      : reporter_(reporter), config_(config), pcs_(pcs) {}
+
+  F Run(Vendor vendor, const math::RowMajorMatrix<F>& input) {
+    size_t max_degree = static_cast<size_t>(input.rows());
+    std::vector<size_t> degrees;
+    for (size_t d = max_degree >> (config_.input_num() - 1); d <= max_degree;
+         d <<= 1) {
+      degrees.push_back(d);
+    }
```

Comment:
Please make private method for this and use this for both `Run()` and `RunExternal()`.